application/classes/user.py

```diff
@@ -13,52 +14,50 @@ class User:
-        self.__name = name
-        self.__code = code
-        self.__dob = dob
-        self.__role_id = role_id
-        self.__country_id = country_id
-        self.__username = username
-        self.__password = password
+    def __init__(self):
+        self.__name = self.__name
+        self.__code = self.__code
+        self.__dob = self.__dob
+        self.__role_id = self.__role_id
+        self.__country_id = self.__country_id
+        self.__username = self.__username
+        self.__password = self.__password
         self.__uuid = uuid.uuid4()
-        self.__created_at = None
-        self.__updated_at = None
-        self.__last_login_at = None
+        
 
         # initialise instance of DBManager
-        self.db_manager = DBManager()
+        self.__db_manager = DBManager()
 
         # set current_time to now
-        self.current_time = datetime.datetime.now()
-
-    def add_user(self):
-        # update created_at and updated_at attributes
-        if self._created_at is None:
-            self._created_at = self.current_time
-        self._updated_at = self.current_time
+        self.__current_time = datetime.datetime.now()
 
+    def add_user(self, name, code, dob, role_id, country_id, username, password):
         # perform database query to save user attributes.
         query = "INSERT INTO users (uuid, name, code, dob, role_id, \
-            country_id, username, password) VALUES ?, ?, ?, ?, ?, ?, ?, ?)"
-        values = (str(self._uuid), self._name, self._code, self._dob, self._role_id,
-                  self._country_id, self._username, self._password, self._created_at)
+            country_id, username, password, updated_at) VALUES ?, ?, ?, ?, ?, ?, ?, ?, ?)"
+        values = (str(uuid.uuid4()), name, code, dob, role_id,
+                  country_id, username, password, self.__current_time)
 
         # call do_insert method from DBmanager.
-        self.db_manager.do_insert(query, [values], dry=False)
+        self.__db_manager.do_insert(query, [values], dry=False)
 
     def update_user(self):
         # update the 'updated_at' attribute.
-        self._updated_at = self.current_time
+        self.__updated_at = self.__current_time
 
         # perform database query to update user attributes.
         query = "UPDATE users SET name=?, code=?, dob=?, role_id=?, \
             country_id=?, username=?, password=?, updated_at=? WHERE uuid=?"
-        values = (self._name, self._code, self._dob, self._role_id, self._country_id,
-                  self._username, self._password, self._updated_at, str(self._uuid))
+        values = (self.__name, self.__code, self.__dob, self.__role_id, self.__country_id,
+                  self.__username, self.__password, self.__updated_at, str(self.__uuid))
 
         # call do_update method from DBmanager.
-        self.db_manager.do_update(query, values)
+        self.__db_manager.do_update(query, values)
 
     def delete_user(self):
```

Comment:
this need to accept a parameter so we can input the information required to identify a user, so that they get deleted.

Reply:
I amended to pass in uuid